packages/atomic/cypress/integration/result-list/result-components/result-multi-value-text.cypress.ts

```diff
@@ -0,0 +1,314 @@
+import {
+  generateComponentHTML,
+  TagProps,
+  TestFixture,
+} from '../../../fixtures/test-fixture';
+import {addFacet} from '../../facets/facet/facet-actions';
+import {FacetSelectors} from '../../facets/facet/facet-selectors';
+import {addResultList, buildTemplateWithSections} from '../result-list-actions';
+import {
+  assertShouldRenderValues,
+  assertDisplaysXMoreLabel,
+  assertDoesNotDisplayXMoreLabel,
+} from './result-multi-value-text-assertions';
+import {
+  resultMultiValueTextComponent,
+  ResultMultiValueTextSelectors,
+} from './result-multi-value-text-selectors';
+import * as CommonFacetAssertions from '../../facets/facet-common-assertions';
+
+export interface MultiValueTextProps {
+  field?: string | number;
+  'max-values-to-display'?: number;
+}
+
+const addMultiValueText = (
+  props: MultiValueTextProps = {},
+  slot: HTMLElement[] = []
+) => {
+  const multiValueTextEl = generateComponentHTML(
+    resultMultiValueTextComponent,
+    props as TagProps
+  );
+  slot.forEach((el) => multiValueTextEl.appendChild(el));
+  return addResultList(
+    buildTemplateWithSections({bottomMetadata: multiValueTextEl})
+  );
+};
+
+export interface FacetWithResponseProps {
+  facetId: string;
+  field: string;
+  values: string[];
+  selectedValues: string[];
+}
+
+const addFacetWithResponse =
+  ({facetId, field, values, selectedValues}: FacetWithResponseProps) =>
+  (fixture: TestFixture) => {
+    fixture
+      .with(addFacet({'facet-id': facetId, field}))
+      .withCustomResponse((response) => {
```

Comment:
Why are we modifying the response? Versus using cypress tests as e2e tests?

For example, on line 61, why not select the values through the browser for real? Is it adding too much time to the test?

Reply:
Because I want to test fields with a total amount of values all the way from 1 to 5, and this seems like the most consistent way to do it.

Reply:
To add some precision @samisayegh : 

It's still doing a query to the API, it's just that sometimes we need to tamper with the results client side to ensure we always get the same response so that the tests don't start to get flaky in a couple months when the index content changes over time.

Granted, I think in a vacuum it would be better to only rely on purely data that comes from the real API/index so as to not hide bugs that would only happen with a full integration.

But for pragmatic reason, I don't find this too bad, and ensuring that we always get consistent results from the index 2 years from now with no content change is... quite a bit of extra work.

If we start to see that there's bugs that start appearing that our test suite was hiding because of the client side tampering we do, we should re-evaluate.